Cython/Compiler/Optimize.py

```diff
@@ -4424,6 +4424,8 @@ def _multiply_string(self, node, string_node, multiplier_node):
                 string_node.unicode_value = encoded_string(
                     string_node.unicode_value * multiplier,
                     string_node.unicode_value.encoding)
+            if not string_node.value.is_unicode:
+                build_string = bytes_literal
```

Comment:
I don't think I understand exactly what's needed from [these two comments](https://github.com/cython/cython/pull/3952#discussion_r564766368) on the other PR. So I haven't made the changes.

Reply:
Actually - I think I do understand so I've made the changes